### PR TITLE
Added status code and its description () for Google API call errors

### PR DESCRIPTION
**Pull request by VitaliiTihobrazov-Melsoft**
If the API call throws an ApiException:
- "error_status" will be set to apiException.getStatusCode()
- in "error", a description of the status code will be added

### CODE REVIEW

File: gpgs/src/java/com/defold/gpgs/GpgsJNI.java [hunk 1]
```diff
@@ -171,6 +170,32 @@ private void sendSimpleMessage(int msg, String key_1, int value_1, String key_2,
         gpgsAddToQueue(msg, message);
     }
 
+    private void sendFailedMessage(int msg, String error_text, Exception e) {
+        if(e != null) {
+
+            if (e instanceof ApiException) {
+                ApiException apiException= (ApiException) e;
```
Comment (britzl) [inline gpgs/src/java/com/defold/gpgs/GpgsJNI.java:177]: Teeny tiny nitpick: Missing a space before the equals sign :-)

File: gpgs/src/java/com/defold/gpgs/GpgsJNI.java [hunk 2]
```diff
@@ -171,6 +170,32 @@ private void sendSimpleMessage(int msg, String key_1, int value_1, String key_2,
...
+                Integer errorStatusCode = apiException.getStatusCode();
+                error_text += ": " + GamesClientStatusCodes.getStatusCodeString(errorStatusCode) +" ("+errorStatusCode.toString()+")";
+
+                sendSimpleMessage(msg,
+                            "status", STATUS_FAILED,
+                            "error_status", errorStatusCode,
```
Comment (britzl) [inline gpgs/src/java/com/defold/gpgs/GpgsJNI.java:183]: Will this be used for logging purposes only or do we anticipate use of the error_status in game code? And if so, should we also expose the status codes as constants defined in gpgs.*?

I think these are the types of status codes you can get right?

https://developers.google.com/android/reference/com/google/android/gms/common/api/CommonStatusCodes

**Review by britzl — CHANGES_REQUESTED** [review]
Looks good. I have a question about how the error_status will be used though.

File: gpgs/src/java/com/defold/gpgs/GpgsJNI.java (hunk 2 above)
Comment (VitaliiTihobrazov-Melsoft) [reply gpgs/src/java/com/defold/gpgs/GpgsJNI.java:183]: Now we use codes for registration purposes. But, probably in the future, we will process codes in code too.
Which ones we will use is not yet known. Adding all codes ahead (as constants defined in gpgs) seems like overkill

Some of the codes are provided here - CommonStatusCodes (https://developers.google.com/android/reference/com/google/android/gms/common/api/CommonStatusCodes)
Another codes is here - GamesClientStatusCodes (https://developers.google.com/android/reference/com/google/android/gms/games/GamesClientStatusCodes.html)
And in other places like - SafetyNetStatusCodes (https://developers.google.com/android/reference/com/google/android/gms/safetynet/SafetyNetStatusCodes)

I propose to postpone the solution of this issue for later

File: gpgs/src/java/com/defold/gpgs/GpgsJNI.java (hunk 1 above)
Comment (VitaliiTihobrazov-Melsoft) [reply gpgs/src/java/com/defold/gpgs/GpgsJNI.java:177]: Sorry I missed )

File: gpgs/src/java/com/defold/gpgs/GpgsJNI.java (hunk 2 above)
Comment (britzl) [reply gpgs/src/java/com/defold/gpgs/GpgsJNI.java:183]: I agree. I was just curious to learn how it was supposed to be used now. Let's postpone until later.

File: gpgs/api/gpgs.script_api [hunk 3]
```diff
@@ -281,6 +281,8 @@
                     - `gpgs.ERROR_STATUS_SNAPSHOT_FOLDER_UNAVAILABLE`
 
                     - `gpgs.ERROR_STATUS_SNAPSHOT_NOT_FOUND`
+
```
Comment (britzl) [inline gpgs/api/gpgs.script_api:284]: The indentation here looks weird. Could you fix this please and then it's ready to merge!

**Review by britzl — CHANGES_REQUESTED** [review]
Indentation is off in the .script_api

**Review by britzl — APPROVED** [review]
Thank you!

**Pull request merged**